File: src/Api/Model/DB/Socio.cs
using Newtonsoft.Json;
using Realms;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CriadorBaseDados.Model.DB
{
    [JsonObject(MemberSerialization.OptIn)]
    public class Socio : RealmObject
    {
       // [JsonProperty]
        [Backlink(nameof(SocioEmpresa.Socio))]
        public IQueryable<SocioEmpresa> SocioEmpresas { get;}
        [JsonProperty]
        [Indexed]
        public string Nome { get; set; }
        [JsonProperty]
        [Indexed]
        public string CNPJ_CPF { get; set; }
        [Ignored]
        [JsonProperty]
        public int CodTipo => Tipo.Cod;
        public TipoSocio Tipo { get; set; }
    }
}

File: src/Mobile/b2b/lib/model/core/socio.dart
import 'socio_empresa.dart';

class Socio {
  final List<SocioEmpresa> socioEmpresas;
  final String nome;
  final String cnpjCpf;
  final int codTipo;

  Socio(this.socioEmpresas, this.nome, this.cnpjCpf, this.codTipo);

  Socio.fromJson(Map<String, dynamic> json)
      : nome = json['nome'],
        cnpjCpf = json['cnpjCpf'],
        codTipo = json['codTipo'],
        socioEmpresas = json['socioEmpresas'] ?? List.empty(growable: true);

  Map<String, dynamic> toMap() => <String, dynamic>{
        'nome': nome,
        'cnpjCpf': cnpjCpf,
        'codTipo': codTipo,
      };
}
